dtfs-central-api/src/v1/controllers/utilisation-report-service/post-keying-data.controller/index.ts

```diff
@@ -16,7 +16,7 @@ export const postKeyingData = async (req: PostKeyingDataRequest, res: Response)
   const { user } = req.body;
 
   try {
-    const feeRecordsAtMatchStatus = await FeeRecordRepo.findByReportIdAndStatusesWithReport(Number(reportId), ['MATCH']);
+    const feeRecordsAtMatchStatus = await FeeRecordRepo.findByReportIdAndStatusesWithReportAndPayments(Number(reportId), ['MATCH']);
```

Comment:
This now loads payments on the fee records as, when updating the join table as part of the keying data generation, we first need to construct the fee record payment groups so that we can split the payments. If you do this again in the event handler, you would have to load every fee record again with all payments attached, which feels like a case where it makes more sense to do it as part of the initial repo call